alkira/resource_alkira_fortinet.go

```diff
@@ -0,0 +1,247 @@
+					"deployed when auto-scale is enabled. Note that auto-scale is not supported " +
+					"with Fortinet at this time. max_instance_count must be greater than or " +
+					"equal to min_instance_count.",
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+			"min_instance_count": {
+				Description: "The minimum number of Fortinet Firewall instances that should be " +
+					" deployed at any point in time.",
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  0,
+			},
+			"name": {
+				Description: "Name of the Fortinet Firewall service.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"segment_names": {
+				Description: "Names of segments associated with the service.",
+				Type:        schema.TypeList,
+				Required:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+			"size": {
+				Description:  "The size of the service, one of `SMALL`, `MEDIUM`, `LARGE`.",
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringInSlice([]string{"SMALL", "MEDIUM", "LARGE"}, false),
+			},
+			"tunnel_protocol": {
+				Description:  "Tunnel Protocol, default to `IPSEC`, could be either `IPSEC` or `GRE`.",
+				Type:         schema.TypeString,
+				Optional:     true,
+				Default:      "IPSEC",
+				ValidateFunc: validation.StringInSlice([]string{"IPSEC", "GRE"}, false),
+			},
+			"version": {
+				Description: "The version of the Fortinet Firewall.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+		},
+	}
+}
+
+func resourceFortinetCreate(d *schema.ResourceData, m interface{}) error {
+	client := m.(*alkira.AlkiraClient)
+
+	request, err := generateFortinetRequest(d, m)
+
+	if err != nil {
+		return err
+	}
+
+	log.Printf("[INFO] Creating fortinet %s", d.Id())
+	id, err := client.CreateFortinet(request)
+
+	if err != nil {
+		return err
+	}
+
+	d.SetId(id)
+	return resourceFortinetRead(d, m)
+
+}
+
+func resourceFortinetRead(d *schema.ResourceData, m interface{}) error {
+	client := m.(*alkira.AlkiraClient)
+
+	f, err := client.GetFortinetById(d.Id())
+
+	if err != nil {
+		return err
+	}
+
+	d.Set("auto_scale", f.AutoScale)
+	d.Set("billing_tag_ids", f.BillingTags)
+	d.Set("credential_id", f.CredentialId)
+	d.Set("cxp", f.Cxp)
+	d.Set("license_type", f.LicenseType)
+	d.Set("management_server_ip", f.ManagementServer.IpAddress)
+	d.Set("management_server_segment", f.ManagementServer.Segment)
+	d.Set("max_instance_count", f.MaxInstanceCount)
+	d.Set("min_instance_count", f.MinInstanceCount)
+	d.Set("name", f.Name)
+	d.Set("segment_names", f.Segments)
+	d.Set("size", f.Size)
+	d.Set("tunnel_protocol", f.TunnelProtocol)
+	d.Set("version", f.Version)
+
+	var instances []map[string]interface{}
+
+	for _, instance := range f.Instances {
+		i := map[string]interface{}{
+			"name":          instance.Name,
+			"serial_number": instance.SerialNumber,
+			"credential_id": instance.CredentialId,
```

Comment:
This `credential_id` is the same as the top level one? if they are the same, why we need two credentials...